source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -257,19 +257,32 @@ void EnvoyQuicServerStream::maybeDecodeTrailers() {
 }
 
 bool EnvoyQuicServerStream::OnStopSending(quic::QuicRstStreamErrorCode error) {
+  // Only called in IETF Quic to close write side.
+  ENVOY_STREAM_LOG(debug, "received STOP_SENDING with reset code={}", *this, error);
+  stats_.rx_reset_.inc();
+  bool end_stream_encoded = local_end_stream_;
+  // This call will close write.
   bool ret = quic::QuicSpdyServerStreamBase::OnStopSending(error);
-  if (read_side_closed()) {
+  if (read_side_closed() && !end_stream_encoded) {
+    // If both directions are closed but end stream hasn't be encoded yet, notify reset callbacks.
```

Comment:
I think you're making an assumption about the value of write_side_closed at this point.  It may be worth an `ASSERT(write_side_closed());` before the `if` statement.

Reply:
done